index.js

```diff
@@ -2,52 +2,85 @@
+const runTests = require('./runTests');
+const { createLoader, validateEmail, validatePassword } = require('./utils');
+const { getToken, sendProgressToApi } = require('./api-service');
 
 const questions = [
   {
     type: 'text',
     name: 'email',
     message: 'Tu correo:',
-    validate: (email) => {
-      if (!email) {
-        return 'Por favor indica tu correo';
-      }
-      if (!validator.validate(email)) {
-        return 'Por favor indica tu correo válido';
-      }
-      return true;
-    },
+    validate: validateEmail,
   },
   {
     type: 'password',
     name: 'password',
     message: 'Tu contraseña',
-    validate: (password) => (!password ? 'Por favor indica tu contraseña' : true),
+    validate: validatePassword,
   },
 ];
 
+const buildProgress = (testsData) => {
+  const {
+    unitId,
+    partId,
+    exerciseId,
+    typeContent,
+  } = laboratoria;
+
+  const progressData = {
+    unitId,
+    partId,
+    exerciseId,
+    type: typeContent,
+    preworkType: 'redesign-prework-fe',
+    progress: {
+      testResults: {
+        ...testsData,
+      },
+      updatedAt: new Date(),
+      /* Add completedAt property only if tests passed
+          to save in BigQuery */
+      ...(testsData.state === 'PASS' && { completedAt: new Date() }),
+    },
+  };
+
+  return progressData;
+};
+
 function main() {
-  prompts(questions)
-    .then((response) => {
-      if (response.email && response.password) {
-        let i = 0;
-        const msg = kleur.bold().italic('Registrando progreso');
-        const spinner = cliSpinners.dots;
-        const interval = setInterval(() => {
-          const { frames } = spinner;
-          // eslint-disable-next-line no-plusplus
-          logUpdate(`${frames[i = ++i % frames.length]} ${msg}`);
-        }, spinner.interval);
-
-        setTimeout(() => {
-          clearInterval(interval);
-          console.log(kleur.green().bold('Listo!'));
-        }, 1500);
-      }
+  return runTests()
+    .then((testsData) => {
+      console.log(kleur.bold().italic('Se terminó de correr los tests. A continuación completa los siguientes datos:'));
+      return prompts(questions)
+        .then((response) => {
+          if (response.email && response.password) {
+            const msg = kleur.bold().italic('Registrando progreso');
+            /* start spinner animation */
+            const interval = createLoader(msg);
+            const progress = buildProgress(testsData);
+            return getToken(response.email, response.password)
+              .then((token) => sendProgressToApi(progress, token))
+              .finally(() => {
+                /* stop spinner animation */
+                clearInterval(interval);
+              });
+          }
+          throw new Error('El correo y contraseña ingresados no son válidos');
+        });
     });
 }
 
-main();
+main()
+  .then((successMessage) => console.log(kleur.green().bold(successMessage)))
```

Comment:
@Danielalab cuando corro los tests obtengo un `Cannot read property 'then' of undefined` proveniente de la linea 60.
Creo q hay algo en el mock de `prompts` q no esta funcionando como se espera.

Para aislar el problema puedes poner un `fit` a cualquiera de los siguientes tests de `index.spec.js/main`
- should be return an error message with wrong email & password
- skipped should be return a success message

Para ejecutar solo `index.spec.js` puedes hacerlo con este comando:
`npm run test -- tests/index.spec.js`